src/PlayAudioUsingMediaStreams/Program.cs

```diff
@@ -1,10 +1,20 @@
+using Microsoft.AspNetCore.HttpOverrides;
 using PlayAudioUsingMediaStreams.WebApi.Services;
 
 var builder = WebApplication.CreateBuilder(args);
 
 // Add services to the container.
 
+builder.Services.AddSpeechClient();
+builder.Services.AddTransient<SoundService>();
+builder.Services.AddTransient<SpeechRecognitionService>();
+
+// For load balancers, reverse proxies, and tunnels like ngrok and VS dev tunnels
+// Follow guidance to secure here: https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer
+builder.Services.Configure<ForwardedHeadersOptions>(options => options.ForwardedHeaders = ForwardedHeaders.All);
```

Comment:
For some reason this isn't necessary for ngrok, but for VS dev tunnels it is.
This ensures that the correct host, scheme, etc is set on the HttpContext.Request object, instead of the localhost one.